### PR TITLE
Accessibility Related changes

### PR DESCRIPTION
**Pull request by prashelke**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [x] PR description and commit message should describe the changes done in this PR
- [x] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [x] Latest Code from master or specific release branch is merged to your branch
- [x] No unwanted\commented\junk code is included
- [x] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [x] Verify that changes are compatible with all relevant browsers and platforms.
- [x] After creating pull request there should not be any conflicts
- [ ] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **New Features**
	- Enhanced accessibility testing features with new configuration options for standards, rules, and severity levels.
	- Added support for web accessibility testing using `TWP.Selenium.Axe.Html` and `Selenium.Axe` packages.
- **Improvements**
	- Updated UI elements and interactions on the Accessibility Testing edit page for a better user experience.
	- Enhanced logic for handling accessibility testing configurations, including severity level adjustments and rule management.
- **Refactor**
	- Refactored code related to multi-select combo boxes and accessibility rule data handling for improved performance and maintainability.
- **Style**
	- Adjusted font weight for toggle buttons from "Bold" to "Regular" for consistency in UI design.

### CODE REVIEW

File: Ginger/GingerCoreCommon/External/Configurations/AccessibilityConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+        {
+            get
+            {
+                return _items;
+            }
+            set
+            {
+                _items = value;
+
+            }
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/GingerCoreCommon/External/Configurations/AccessibilityConfiguration.cs:82]: Done

File: Ginger/GingerCoreCommon/External/Configurations/AccessibilityConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+        {
+            get
+            {
+                return _SeverityItems;
+            }
+            set
+            {
+                _SeverityItems = value;
+
+            }
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/GingerCoreCommon/External/Configurations/AccessibilityConfiguration.cs:115]: done

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs [hunk 3]
```diff
@@ -1620,6 +1612,9 @@ private void ActAccessibility(ActAccessibilityTesting act)
                 {
                     axeResult = axeBuilder.Analyze();
                 }
+                 //string path = $"{WorkSpace.Instance.Solution.ContainingFolderFullPath}\\AxeReport.html";
+                //Driver.CreateAxeHtmlReport(axeResult, path);
+                //act.AddOrUpdateReturnParamActual(ParamName: "Accessibility report", ActualValue: path );
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs]: Updated

File: Ginger/GingerCoreCommon/External/Configurations/AccessibilityConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+                {
+                    mSeverityOperationValueList = value;
+                    OnPropertyChanged(nameof(SeverityOperationValueList));
+                }
+            }
+        }
+
+
+
+        public override string ItemName { get => throw new System.NotImplementedException(); set => throw new System.NotImplementedException(); }
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/GingerCoreCommon/External/Configurations/AccessibilityConfiguration.cs:138]: Handled

File: Ginger/GingerCoreCommon/External/Configurations/AccessibilityConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+            accessibilityRules.Add( new AccessibilityRuleData { Active = false, RuleID = "table-duplicate-name", Description = "Ensure the <caption> element does not contain the same text as the summary attribute", Impact = "Minor", Tags = "bestpractice" ,ACTRules = ""});
+            accessibilityRules.Add( new AccessibilityRuleData { Active = false, RuleID = "color-contrast-enhanced", Description = "Ensures the contrast between foreground and background colors meets WCAG 2 AAA enhanced contrast ratio thresholds", Impact = "Serious", Tags = "wcag2aaa" ,ACTRules = "09o5cg"});
+            accessibilityRules.Add( new AccessibilityRuleData { Active = false, RuleID = "identical-links-same-purpose", Description = "Ensure that links with the same accessible name serve a similar purpose", Impact = "Minor", Tags = "wcag2aaa" ,ACTRules = "b20e66"});
+            accessibilityRules.Add( new AccessibilityRuleData { Active = false, RuleID = "meta-refresh-no-exceptions", Description = "Ensures < meta http - equiv = refresh > is not used for delayed refresh", Impact = "Minor", Tags = "wcag2aaa", ACTRules = "bisz58"});
+            return accessibilityRules;
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/GingerCoreCommon/External/Configurations/AccessibilityConfiguration.cs:235]: handled

File: Ginger/Ginger/UserControlsLib/MultiSelectComboBox.xaml.cs [hunk 6]
```diff
@@ -209,8 +210,21 @@
...
+                        {
+                            SelectedItems.Add(new OperationValues() { Value = node.Title, Guid = Guid.Parse(this.ItemsSource[node.Title].ToString()) });
+                            temp.Add(new OperationValues() { Value = node.Title, Guid = Guid.Parse(this.ItemsSource[node.Title].ToString()) });
+                        }
+                        else
+                        {
+                            SelectedItems.Add(new OperationValues() { Value = node.Title });
+                            temp.Add(new OperationValues() { Value = node.Title });
+                        }
+                    }
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/Ginger/UserControlsLib/MultiSelectComboBox.xaml.cs]: Handled

File: Ginger/GingerCoreCommon/VariablesLib/OperationValues.cs [hunk 7]
```diff
@@ -45,6 +45,24 @@ public string Value
...
+            }
+            set
+            {
+                if(mDisplayName != value)
+                {
+                    mDisplayName = value;
+                    OnPropertyChanged(nameof(value));
+                }
+            }
+        }
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/GingerCoreCommon/VariablesLib/OperationValues.cs]: Handled

File: Ginger/GingerCoreNET/ActionsLib/UI/Web/ActAccessibilityTesting.cs [hunk 8]
```diff
@@ -199,5 +315,120 @@
...
+
+    //    public void OnPropertyChanged(string name)
+    //    {
+    //        PropertyChangedEventHandler handler = PropertyChanged;
+    //        if (handler != null)
+    //        {
+    //            handler(this, new PropertyChangedEventArgs(name));
+    //        }
+    //    }
+    //}
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/GingerCoreNET/ActionsLib/UI/Web/ActAccessibilityTesting.cs]: Removed

File: Ginger/GingerCoreCommon/External/Configurations/AccessibilityRuleData.cs [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        public override string ItemName { get; set; }
+
+        public void OnPropertyChanged(string name)
+        {
+            PropertyChangedEventHandler handler = PropertyChanged;
+            if (handler != null)
+            {
+                handler(this, new PropertyChangedEventArgs(name));
+            }
+        }
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/GingerCoreCommon/External/Configurations/AccessibilityRuleData.cs]: Added

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs [hunk 10]
```diff
@@ -1635,23 +1625,50 @@
...
+            {
+                List<string> sevritylist = act.SeverityOperationValueList.Select(x => x.Value.ToLower()).ToList();
+                List<string> Violationsevrity = axeResult.Violations.Any() ? axeResult.Violations.Select(x => x.Impact.ToLower()).ToList() : new List<string>();
+                foreach(string severity in sevritylist)
+                {
+                    hasAnyViolations = !Violationsevrity.Any(y => y.Equals(severity));
+                }
+                
+            }
+            //bool hasAnyViolations = axeResult.Violations.Any();
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs]: Updated

File: Ginger/GingerCoreNET/ActionsLib/UI/Web/ActAccessibilityTesting.cs [hunk 11]
```diff
@@ -199,5 +315,91 @@
...
+            else
+            {
+                RulesItemsdata = RulesItemsdata == null ? GetRuleList() : RulesItemsdata;
+            }
+            foreach(AccessibilityRuleData ruleData in Exculedrules)
+            {
+                ruleData.Active = ExcludeRuleList!= null && ExcludeRuleList.Any(x=>x.RuleID == ruleData.RuleID) ? ExcludeRuleList.FirstOrDefault(x => x.RuleID == ruleData.RuleID).Active : false;
+            }
+            return Exculedrules.Any() ? new ObservableList<AccessibilityRuleData>(Exculedrules) : RulesItemsdata;
+        }
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/GingerCoreNET/ActionsLib/UI/Web/ActAccessibilityTesting.cs]: Updated

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**